hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -678,91 +497,46 @@ def should_generate_more():
                 self.first_bug_found_at + 1000, self.last_bug_found_at * 2
             )
 
+        # GenerationParameters are a set of decisions we make that are global
+        # to the whole test case, used to bias the data generation in various
+        # ways. This is an approach very very loosely inspired by the paper
+        # "Swarm testing." by Groce et al. in that it induces deliberate
+        # correlation between otherwise independent decisions made during the
+        # generation process.
```

Comment:
Closes #1637 then?

Reply:
Not yet. It's a big step towards being able to do so, but it's not quite there yet. The thing it would need to close #1637 is a bit more context awareness - if it could pick the alphabet based on the current strategy being used then it would count as swarm testing in my books, but right now it doesn't know the difference between the bits drawn for say a a `boolean()` and a `one_of(a, b)`